pkg/utils/clients/factories.go

```diff
@@ -38,6 +40,11 @@ func NewHTTPMessageReceiver(port Port) *kncloudevents.HTTPMessageReceiver {
 	return kncloudevents.NewHTTPMessageReceiver(int(port))
 }
 
+// NewHTTPMessageReceiverWithChecker wraps kncloudevents.NewHttpMessageReceiver with a authentication checker.
+func NewHTTPMessageReceiverWithChecker(port Port, authType authcheck.AuthType) *kncloudevents.HTTPMessageReceiver {
```

Comment:
Is it possible to unit test this?

I _think_ it is possible by passing port = 0, which should mean find any available port. And a way to inject/overwrite the auth check implementation that is used.

I probably won't hold up the PR if this turns out to be too onerous, but I would feel much more comfortable if this was unit tested.

Reply:
I was also commenting on testing this, but more in adding some check in our e2e test.

Reply:
E2E test would be even better, but likely difficult because we only have a single Broker data plane and testing this would require us to break it, thus the other tests could not run in parallel with this one.

Reply:
I added UTs for this. I am trying to add E2E test for brokercell currently, but as Adam mentioned, it has some difficulty as we only have a single broker data plane (I probably will add e2e test before we run all sub-tests.)